### DISCUSSION TITLE
Multicascader: checked status problem

### DISCUSSION THREAD

**Discussion by zuosiliang** [rsuite/rsuite#1737] [General]
consider the following source data giving to multicascader, if I check on the node(id 15818), the node (Id 14888) will be indeterminate status which should be fully checked since its only child that is 15818 is checked.
`
[
    {
      id: 15818,
      idPrefix: true,
      suffix: 1,
      rawId: 15818,
      name: '全量表标签',
      status: 'ENABLE',
      parentStatus: 'ENABLE',
      sharing: false,
      parentSharing: false,
      type: 'CUSTOM',
      ruleType: 'DEFAULT',
      dataType: null,
      path: {
        id: 14888,
        name: '全量表标签',
        status: 'ENABLE',
        sharing: false,
        parentSharing: false,
        children: [],
      },
      conditional: false,
      children: [],
      isSystemDirectory: false,
      parent: null,
    },
    {
      id: 14888,
      idPrefix: true,
      suffix: 1,
      rawId: 14888,
      name: '全量表标签',
      status: 'ENABLE',
      parentStatus: 'ENABLE',
      sharing: false,
      parentSharing: false,
      type: 'DIRECTORY',
      ruleType: null,
      dataType: null,
      path: null,
      conditional: false,
      children: [
        {
          id: 15818,
          originId: 15818,
          name: '全量表标签',
          status: 'ENABLE',
          parentStatus: 'ENABLE',
          sharing: false,
          parentSharing: false,
          type: 'CUSTOM',
          ruleType: 'DEFAULT',
          dataType: null,
          path: {
            id: 14888,
            name: '全量表标签',
            status: 'ENABLE',
            sharing: false,
            parentSharing: false,
            children: [],
          },
          conditional: false,
          children: null,
          isSystemDirectory: false,
          parent: {
            id: 14888,
            idPrefix: true,
            suffix: 1,
            rawId: 14888,
            name: '全量表标签',
            status: 'ENABLE',
            parentStatus: 'ENABLE',
            sharing: false,
            parentSharing: false,
            type: 'DIRECTORY',
            ruleType: null,
            dataType: null,
            path: null,
            conditional: false,
            isSystemDirectory: false,
            parent: null,
          },
        },
      ],
      isSystemDirectory: false,
      parent: null,
    },
  ];
`

**Comment by javidmiandadsb** [comment]
May i know the solution, same problem facing?

**Comment by simonguo** [comment]
Multicascader's data is a tree structure. The key of each tree node must be unique.
